corehq/apps/reports/standard/users/reports.py

```diff
@@ -179,15 +180,39 @@ def _html_list(changes, unstyled=True):
 
     properties = UserHistoryReport.get_primary_properties(domain)
     properties.pop("user_data", None)
-    primary_changes = {
-        properties.get(key, key): value for key, value in changes.items()
-        if key in properties
-    }
-    more_count = len(changes) - len(primary_changes)
+    primary_changes = {}
+
+    for key, value in changes.items():
+        if key in properties:
+            if key == 'commcare_location_id':
+                primary_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
+            if key == 'location_id':
+                primary_changes["location"] = cached_location_id_to_name(value)
+            else:
+                primary_changes[key] = value
+
+    all_changes = {}
+
+    for key, value in changes.items():
+        if key == 'user_data':
+            for key, value in changes['user_data'].items():
+                if key == 'commcare_location_id':
+                    all_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
+                if key == 'location_id':
+                    all_changes["location"] = cached_location_id_to_name(value)
+                else:
+                    all_changes[key] = value
+        if key == 'commcare_location_id':
+            all_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
+        if key == 'location_id':
+            all_changes["location"] = cached_location_id_to_name(value)
+        else:
+            all_changes[key] = value
 
+    more_count = len(all_changes) - len(primary_changes)
     return render_to_string("reports/standard/partials/user_history_changes.html", {
         "primary_changes": _html_list(primary_changes) if primary_changes else None,
-        "all_changes": _html_list(changes, unstyled=False),
+        "all_changes": _html_list(all_changes, unstyled=True) if all_changes else None,
```

Comment:
All usages of `_html_list` have `unstyled=True` now, don't they? If so, you can remove the parameter altogether.